src/Okta.Sdk.IntegrationTests/NetworkZoneScenarios.cs

```diff
@@ -0,0 +1,134 @@
+            NetworkZone newZone = BuildNetworkZoneObject();
+            var createZoneResponse = await zonesClient.CreateNetworkZoneAsync(newZone);
+            try
+            {
+                createZoneResponse.Id.Should().NotBeNullOrEmpty();
+                createZoneResponse.Name.Should().Be(newZone.Name);
+                createZoneResponse.Type.Should().Be(newZone.Type);
+                createZoneResponse.Status.Should().Be(newZone.Status);
+                createZoneResponse.Gateways.Should().HaveCount(2);
+                createZoneResponse.Proxies.Should().HaveCount(2);
+
+                createZoneResponse.Name = "Another Name";
+                createZoneResponse.Gateways.Add(new NetworkZoneAddress
+                {
+                    Type = NetworkZoneAddressType.Cidr,
+                    Value = "1.1.1.1/24",
+                });
+
+                var updateZoneResponse = await zonesClient.UpdateNetworkZoneAsync(createZoneResponse, createZoneResponse.Id);
+                updateZoneResponse.Name.Should().Be(createZoneResponse.Name);
+                updateZoneResponse.Gateways.Count.Should().Be(3);
+            }
+            finally
+            {
+                await zonesClient.DeleteNetworkZoneAsync(createZoneResponse.Id);
+            }
+        }
+
+        [Fact]
+        public async Task DeactivateAndActivateNetworkZone()
+        {
+            var oktaClient = TestClient.Create();
+            var zonesClient = oktaClient.NetworkZones;
+            NetworkZone newZone = BuildNetworkZoneObject();
+            var createZoneResponse = await zonesClient.CreateNetworkZoneAsync(newZone);
+            try
+            {
+                var deactivateZoneResponse = await zonesClient.DeactivateNetworkZoneAsync(createZoneResponse.Id);
+                deactivateZoneResponse.Status.Should().Be(NetworkZoneStatus.Inactive);
+                var activateZoneResponse = await zonesClient.ActivateNetworkZoneAsync(deactivateZoneResponse.Id);
+                activateZoneResponse.Status.Should().Be(NetworkZoneStatus.Active);
+            }
+            finally
+            {
+                await zonesClient.DeleteNetworkZoneAsync(createZoneResponse.Id);
+            }
+        }
+
+        [Fact]
+        public async Task ListAndGetNetworkZones()
+        {
+            var oktaClient = TestClient.Create();
+            var zonesClient = oktaClient.NetworkZones;
+            NetworkZone newZone = BuildNetworkZoneObject();
+            var createZoneResponse = await zonesClient.CreateNetworkZoneAsync(newZone);
+            try
+            {
+                var allNetworkZones = zonesClient.ListNetworkZones();
+                var createdZone = await allNetworkZones.FirstOrDefaultAsync(z => z.Id == createZoneResponse.Id);
+
+                createdZone.Should().NotBeNull();
+
+                var getResponse = await zonesClient.GetNetworkZoneAsync(createZoneResponse.Id);
```

Comment:
I think this should be two different tests.  One for `List` and one for `Get`, for reasons previously stated.

Reply:
Agree, I'll split it into two.